maven-embedder/src/main/java/org/apache/maven/cli/transfer/ConsoleMavenTransferListener.java

```diff
@@ -135,7 +136,7 @@ public synchronized void transferFailed(TransferEvent event) {
 
     private void overridePreviousTransfer(TransferEvent event) {
         if (lastLength > 0) {
-            StringBuilder buffer = new StringBuilder(128);
+            StringBuilder buffer = new StringBuilder(lastLength + 1);
```

Comment:
This class is completely (and overly btw) synchronised, so why not using a single `StringBuilder buffer` field and use `setLength(0)` when needed.

Reply:
Yes this is true, I also noticed that the synchronized block at line 71 is not needed since the transferProgressed() method is synchronized already and synchronising on an instance var have the same effect.

Reply:
And also, the `transfers` field does not need to be wrapped in a `Collections.synchronizedMap` collection.

Reply:
Good catch! I will remove the `Collections.synchronizedMap`